### PR TITLE
Multi Target Trx Issue : LogFilePrefix Parameter Added

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
## Description
RFC: https://github.com/microsoft/vstest-docs/pull/204

Adding LogFilePrefix parameter to specify trx name prefix.
When this parameter is specified as `dotnet test --logger:"trx;LogFilePrefix=results"`, the prefix will be appended with target framework and timestamp generating a unique trx file for every target.
Example: `E:\TestResults\results_net451_2018_12-24_14-01-07-176.trx  `

Timestamp is appended with milli seconds precision.
## Related issue
#1603

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/Utilities/RunSettingsUtilities.cs [hunk 1]
```diff
@@ -47,6 +47,23 @@ public static string GetTestResultsDirectory(RunConfiguration runConfiguration)
             return resultsDirectory;
         }
 
+        /// <summary>
+        /// Gets the taregt framework from the run configuration
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/Utilities/RunSettingsUtilities.cs]: Nit: target

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs [hunk 2]
```diff
@@ -443,14 +444,27 @@ private void DeriveTrxFilePath()
         {
             if (this.parametersDictionary != null)
             {
-                var isLogFileNameParameterExists = this.parametersDictionary.TryGetValue(TrxLoggerConstants.LogFileNameKey, out string logFileNameValue);
-                if (isLogFileNameParameterExists && !string.IsNullOrWhiteSpace(logFileNameValue))
+                var isLogFilePrefixParameterExists = this.parametersDictionary.TryGetValue(TrxLoggerConstants.LogFilePrefixKey, out string logFilePrefixValue);
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: Can someone give both logFilePrefix and LogFileName ?

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs [hunk 3]
```diff
@@ -443,14 +444,27 @@ private void DeriveTrxFilePath()
...
+
+                if (isLogFilePrefixParameterExists && !string.IsNullOrWhiteSpace(logFilePrefixValue))              
                 {
-                    this.trxFilePath = Path.Combine(this.testResultsDirPath, logFileNameValue);
+                    var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework] ?? string.Empty;
+                    framework = NuGetFramework.Parse(framework).GetShortFolderName();
+
+                    logFilePrefixValue = logFilePrefixValue.Replace(".trx", string.Empty) + "_" + framework + DateTime.Now.ToString("_yyyyMMddHHmmssfff", DateTimeFormatInfo.InvariantInfo) + ".trx";
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: make .trx as constant

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/ObjectModel/TestRun.cs [hunk 4]
```diff
@@ -191,7 +191,7 @@ private static string FormatDateTimeForRunName(DateTime timeStamp)
         {
             // We use custom format string to make sure that runs are sorted in the same way on all intl machines.
             // This is both for directory names and for Data Warehouse.
-            return timeStamp.ToString("yyyy-MM-dd HH:mm:ss", DateTimeFormatInfo.InvariantInfo);
+            return timeStamp.ToString("yyyy-MM-dd HH:mm:ss:fff", DateTimeFormatInfo.InvariantInfo);
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/ObjectModel/TestRun.cs]: why is this needed?

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs [hunk 5]
```diff
@@ -443,14 +444,27 @@ private void DeriveTrxFilePath()
...
             {
-                var isLogFileNameParameterExists = this.parametersDictionary.TryGetValue(TrxLoggerConstants.LogFileNameKey, out string logFileNameValue);
-                if (isLogFileNameParameterExists && !string.IsNullOrWhiteSpace(logFileNameValue))
+                var isLogFilePrefixParameterExists = this.parametersDictionary.TryGetValue(TrxLoggerConstants.LogFilePrefixKey, out string logFilePrefixValue);
+
+                if (isLogFilePrefixParameterExists && !string.IsNullOrWhiteSpace(logFilePrefixValue))              
                 {
-                    this.trxFilePath = Path.Combine(this.testResultsDirPath, logFileNameValue);
+                    var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework] ?? string.Empty;
+                    framework = NuGetFramework.Parse(framework).GetShortFolderName();
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: can this cause invalid chars in file path?

**Review by mayankbansal018 — CHANGES_REQUESTED** [review]
:clock1:

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs (hunk 2 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: We will throw in logger initialize in such case

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs (hunk 3 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: Done

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs [hunk 6]
```diff
@@ -443,14 +444,27 @@ private void DeriveTrxFilePath()
         {
             if (this.parametersDictionary != null)
             {
```
[the remark this answers is not in the document]
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: Done

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/ObjectModel/TestRun.cs (hunk 4 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.TrxLogger/ObjectModel/TestRun.cs]: Removed

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

**Comment by singhsarab** [comment]
@vagisha-nidhi @hvinett We will need this for the html logger as well, Can you guys sync up on this.
